wire-ios-system/Source/DateProviding/SystemDateProvider.swift

```diff
@@ -0,0 +1,29 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+extension DateProviding where Self == SystemDateProvider {
+    /// Returns a new instance of `SystemDateProvider`.
+    public static var system: Self { .init() }
+}
```

Comment:
Somehow I don't really understand the benefit of this addition `where Self == SystemDateProvider`. I probably would just have written:
`public static var system: SystemDateProvider { .init() }`

Just out of interested: is there any difference?